src/wmtk/io/Cache.cpp

```diff
@@ -195,18 +254,32 @@ void Cache::write_mesh(
     const std::string& name,
     const std::map<std::string, std::vector<int64_t>>& multimesh_names)
 {
-    const auto it = m_file_paths.find(name);
-
-    fs::path p;
-
-    if (it == m_file_paths.end()) {
-        // file does not exist yet --> create it
-        p = create_unique_file(name, ".hdf5");
-        m_file_paths[name] = p;
+    // write to file
+    if (!m_cache_dir.empty()) {
+        const auto it = m_file_paths.find(name);
+        fs::path p;
+        if (it == m_file_paths.end()) {
+            // file does not exist yet --> create it
+            p = create_cache_file(name, ".hdf5");
+            m_file_paths[name] = p;
+        } else {
+            p = it->second;
+        }
+        HDF5Writer writer(p);
+        m.serialize(writer, &m);
     } else {
-        p = it->second;
+        // write in meshes list
+        const auto it = m_meshes.find(name);
+        std::shared_ptr<Mesh> pm = const_cast<Mesh&>(m).shared_from_this();
```

Comment:
this const cast shouldn'lt be necessary, as shared_from_this can return a `std::shared_ptr<const Mesh>` ? this occurs later on as well